tests/cuda/cuda_tensor_02.cu

```diff
@@ -0,0 +1,252 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Test operator[] and norm_square of cuda_tensor.
+
+#include <deal.II/base/tensor.h>
+
+#include "../tests.h"
+
+template <int rank, int dim, typename Number>
+__global__ void
+miscellaneous_kernel()
+{
+  // constructors
+  typename Tensor<rank, dim, Number>::array_type array{};
+  Tensor<rank, dim, Number>                      dummy_1(array);
+  Tensor<rank, dim, Number>                      dummy_2;
+  Tensor<rank, dim, Number>                      dummy_3 = dummy_2;
+
+  // access
+  Tensor<rank + 1, dim, Number> initializer_1;
+  const auto                    dummy_5 = initializer_1[0];
+
+  // assignment
+  dummy_2 = dummy_3;
```

Comment:
This test is really weak. We should check the result.

Reply:
Yes. The only problems I have seen lately is compiling and running. Hence, I didn't bother testing the result to be identical with the host result so far.